packages/block-library/src/gallery/use-get-media.js

```diff
@@ -33,6 +33,7 @@ export default function useGetMedia( innerBlockImages ) {
 			return select( coreStore ).getMediaItems( {
 				include: imageIds.join( ',' ),
 				per_page: -1,
+				orderby: 'include',
```

Comment:
This way, data cache keys are the same, so we re-used data from the store instead of re-fetching it.